core/primitives/src/runtime/in_memory_contract.rs

```diff
@@ -0,0 +1,56 @@
+use crate::{hash::CryptoHash, types::AccountId};
+use near_vm_errors::VMError;
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::sync::RwLock;
+
+#[derive(Clone)]
+pub struct InMemoryContracts(
+    pub  Arc<
+        RwLock<HashMap<AccountId, Option<(CryptoHash, Result<wasmer_runtime::Module, VMError>)>>>,
+    >,
+);
+
+impl std::fmt::Debug for InMemoryContracts {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "InMemoryContracts")
+    }
+}
+
+impl InMemoryContracts {
+    pub fn new(in_memory_account_ids: &Vec<AccountId>) -> Self {
+        let mut map = HashMap::new();
+        for account_id in in_memory_account_ids {
+            map.insert(account_id.clone(), None);
+        }
+        Self(Arc::new(RwLock::new(map)))
+    }
+
+    pub fn get(
+        &self,
+        account_id: &AccountId,
+        key: &CryptoHash,
+    ) -> Option<Option<Result<wasmer_runtime::Module, VMError>>> {
+        match self.0.read().unwrap().get(account_id) {
+            Some(Some((k, r))) => {
+                if k == key {
+                    Some(Some(r.clone()))
+                } else {
+                    None
+                }
+            }
+            Some(None) => Some(None),
+            None => None,
```

Comment:
Why do we return None when there is a contract with another key and when there is no such account_id, but Some(None) when there is no contract for this account? This seem weird

Reply:
it may well be intended, but I think it is at least worth documenting

Reply:
Good catch, it should be Some(None) too. I also add a doc string